Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,111 @@
 
-    [String] [Parameter(Mandatory = $true)]
-    $DeployToSlotFlag,
+try{
 
-    [String] [Parameter(Mandatory = $false)]
-    $ResourceGroupName,
+	# Get inputs.
+	$WebAppName = Get-VstsInput -Name WebAppName -Require
+	$DeployToSlotFlag = Get-VstsInput -Name DeployToSlotFlag -Require -AsBool
+	$ResourceGroupName = Get-VstsInput -Name ResourceGroupName
+	$SlotName = Get-VstsInput -Name SlotName
+	$Package = Get-VstsInput -Name Package -Require
+	$RemoveAdditionalFilesFlag = Get-VstsInput -Name RemoveAdditionalFilesFlag -AsBool
+	$ExcludeFilesFromAppDataFlag = Get-VstsInput -Name ExcludeFilesFromAppDataFlag -AsBool
+	$TakeAppOfflineFlag = Get-VstsInput -Name TakeAppOfflineFlag -AsBool
+	$VirtualApplication = Get-VstsInput -Name VirtualApplication 
+	$AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+	$WebAppUri = Get-VstsInput -Name WebAppUri
+	$SetParametersFile = Get-VstsInput -Name SetParametersFile
 
-    [String] [Parameter(Mandatory = $false)]
-    $SlotName,
+	# Initialize Azure.
 
-    [String] [Parameter(Mandatory = $true)]
-    $Package,
-    
-    [String] [Parameter(Mandatory = $false)]
-    $SetParametersFile,
+	Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+	Initialize-Azure
 
-    [String] [Parameter(Mandatory = $false)]
-    $RemoveAdditionalFilesFlag,
+	# Import the loc strings.
+	Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
 
-    [String] [Parameter(Mandatory = $false)]
-    $ExcludeFilesFromAppDataFlag,
+	$WebAppUri = $WebAppUri.Trim()
+	$Package = "$Package".Trim('"').Trim()
 
-    [String] [Parameter(Mandatory = $false)]
-    $TakeAppOfflineFlag,
-
-    [String] [Parameter(Mandatory = $false)]
-    $VirtualApplication,
-
-	[String] [Parameter(Mandatory = $false)]
-	[String] $AdditionalArguments,
-
-    [String] [Parameter(Mandatory = $false)]
-    [string]$WebAppUri
-)
+	if( [string]::IsNullOrEmpty($Package) ){
+		Throw (Get-VstsLocString -Key "Invalidwebapppackagepathprovided")
+	}
 
-Write-Verbose "Starting AzureRM WebApp Deployment Task"
+	$SetParametersFile = $SetParametersFile.Trim('"').Trim()
 
-Write-Verbose "ConnectedServiceName = $ConnectedServiceName"
-Write-Verbose "WebAppName = $WebAppName"
-Write-Verbose "DeployToSlotFlag = $DeployToSlotFlag"
-Write-Verbose "ResourceGroupName = $ResourceGroupName"
-Write-Verbose "SlotName = $SlotName"
-Write-Verbose "Package = $Package"
-Write-Verbose "SetParametersFile = $SetParametersFile"
-Write-Verbose "RemoveAdditionalFilesFlag = $RemoveAdditionalFilesFlag"
-Write-Verbose "ExcludeFilesFromAppDataFlag = $ExcludeFilesFromAppDataFlag"
-Write-Verbose "TakeAppOfflineFlag = $TakeAppOfflineFlag"
-Write-Verbose "VirtualApplication = $VirtualApplication"
-Write-Verbose "AdditionalArguments = $AdditionalArguments"
-Write-Verbose "WebAppUri = $WebAppUri"
 
-$WebAppUri = $WebAppUri.Trim()
-$Package = $Package.Trim('"').Trim()
+	# Load all dependent files for execution
+	. $PSScriptRoot/AzureUtility.ps1 -Force
```

Comment:
I thought about this a bit more. Beware there are differences between dot sourcing and importing as a module (modules run within a module context with a different variable inheritance hierarchy and separate session state). I was quite confused when I originally saw importing the ps1 file.

If you're comfortable with switching this over to dot sourcing, then all the better. If not, you may want to consider leaving it as is - sometimes I'm hesitant to change something if it's working as is - or consider changing the file names to psm1 so it's not confusing.

It's your call. If you end up leaving it as import-module and ps1 extension, it would be nice to add a comment - or a TODO comment - so it's not confusing to anyone who needs to read the source code in the future.